src/CodeGen_X86.cpp

```diff
@@ -277,13 +248,147 @@ void CodeGen_X86::visit(const Sub *op) {
     }
 }
 
+void CodeGen_X86::visit(const GT *op) {
+    Type t = op->a.type();
+    int bits = t.width * t.bits;
+    if (t.width == 1 || bits % 128 == 0) {
+        // LLVM is fine for native vector widths or scalars
+        CodeGen_Posix::visit(op);
+    } else {
+        // Non-native vector widths get legalized poorly by llvm. We
+        // split it up ourselves.
+        Value *a = codegen(op->a), *b = codegen(op->b);
+
+        int slice_size = 128 / t.bits;
+
+        vector<Value *> result;
+        for (int i = 0; i < op->type.width; i += slice_size) {
+            Value *sa = slice_vector(a, i, slice_size);
+            Value *sb = slice_vector(b, i, slice_size);
+            Value *slice_value;
+            if (t.is_float()) {
+                slice_value = builder->CreateFCmpOGT(sa, sb);
+            } else if (t.is_int()) {
+                slice_value = builder->CreateICmpSGT(sa, sb);
+            } else {
+                slice_value = builder->CreateICmpUGT(sa, sb);
+            }
+            result.push_back(slice_value);
+        }
+
+        value = concat_vectors(result);
+        value = slice_vector(value, 0, t.width);
+    }
+}
+
+void CodeGen_X86::visit(const EQ *op) {
+    Type t = op->a.type();
+    int bits = t.width * t.bits;
+    if (t.width == 1 || bits % 128 == 0) {
+        // LLVM is fine for native vector widths or scalars
+        CodeGen::visit(op);
+    } else {
+        // Non-native vector widths get legalized poorly by llvm. We
+        // split it up ourselves.
+        Value *a = codegen(op->a), *b = codegen(op->b);
+
+        int slice_size = 128 / t.bits;
```

Comment:
Is this correct/efficient for AVX? Seems like it might produce SSE comparisons when AVX would work. I don't know if AVX actually has AVX comparison instructions, but Googling suggests there are vcmp*ps instructions.